### ISSUE TITLE
Decide on a number representation

### ISSUE THREAD

**Issue by cameronfletcher** [LykkeCorp/guidelines#3]
So, in JSON numbers are either an integer or a floating point which has the potential to cause issues.
Should we represent numbers as a string to avoid complications?

**Comment by cameronfletcher** [comment]
Food for thought: https://sendgrid.com/blog/data-types-for-marketers-numbers-string-and-booleans/

**Comment by tarurar** [comment]
I vote for representing numbers as strings. Already had this issue with numbers when integrating with our MarketMaker.

**Comment by cameronfletcher** [comment]
Any objections?  
If not, @tarurar PR?

**Comment by yreynhout-lykke** [comment]
I'd rather first hear what those alleged complications are before jumping to conclusions. What do the numbers represent? Amounts, identifiers, money? I think it's going to be hard to come up with one answer for all cases. Sometimes it makes more sense to ship the metadata with the data, thereby suppressing any ambiguity. Just turning numbers into strings is not going to make all problems go away, it merely shifts some of them towards the consumer.

**Comment by JoeStead** [comment]
Yep, agree with @yreynhout-lykke on this one. We can either deserialise to a number literal (that could fail if it's a float and you expect an int), or we can deserialise to a string, parse the number and face the exact same problem.
